python/services/dataservice/dmod/dataservice/service.py

```diff
@@ -992,4 +852,206 @@ async def perform_checks_for_job(self, job: Job) -> bool:
+
+    async def start(self) -> NoReturn:
+        await self._manage_temp_datasets()
+
+    async def _manage_temp_datasets(self) -> NoReturn:
+        """
+        Async task for managing temporary datasets, including updating expire times and purging of expired datasets.
+        """
+        while True:
+            # Ensure that mark and sweep doesn't proceed while something is potentially to linking datasets
+            while self._tracker.value > 0:
+                await asyncio.sleep(10)
+            self._temp_dataset_mark_and_sweep()
+            await asyncio.sleep(3600)
+
+    def _temp_dataset_mark_and_sweep(self):
+        """
+        Encapsulation of the required behavior for a single iteration through ::method:`manage_temp_datasets`.
+
+        Method scans for in use temporary datasets, updating the expire times of each.  It then deletes any expired
+        datasets marked for deletion already within the instance's private field.  Finally, it updates the instance's
+        private field for marked-for-deletion datasets with any that are now expired (and thus will be removed on the
+        next invocation).
+        """
+        # Get these upfront, since no meaningful changes can happen during the course of an iteration
+        temp_datasets = {ds_name: ds for ds_name, ds in self._managers.known_datasets().items() if ds.is_temporary}
+
+        # Account for any in-use temporary datasets by potentially unmarking and updating expire time
+        for ds in (ds for _, ds in temp_datasets.items() if ds.manager.get_user_ids_for_dataset(ds)):
+            if ds.name in self._marked_expired_datasets:
+                self._marked_expired_datasets.remove(ds.name)
+
+            assert ds.expires is not None
+            ds.extend_life(timedelta(days=1) if ds.expires < (datetime.now()+timedelta(days=1)) else timedelta(hours=1))
+
+        # Delete any datasets previously marked for deletion
+        for ds in (temp_datasets.pop(ds_name) for ds_name in self._marked_expired_datasets):
+            ds.manager.delete(dataset=ds)
+
+        # Mark any expired datasets for deletion on the next iteration
+        self._marked_expired_datasets.update(n for n, ds in temp_datasets.items() if ds.expires > datetime.now())
+
+class DataProvisionManager:
+    """
+    Task method to periodically associate, un-associate, and (when needed) generate required datasets with/for jobs.
+    Start the task by calling ::method:`start`.
+
+    Parameters
+    ----------
+    job_util : JobUtil
+        Access and update jobs
+    dataset_manager_collection : DatasetManagerCollection
+        Facilitates creating and accessing Datasets
+    docker_s3fs_helper : DockerS3FSPluginHelper
+        Facilitates initialize Docker volumes for jobs
+    data_derive_util : DataDeriveUtil
+        Facilitates deriving data and datasets
+    tracker : ActiveOperationTracker
+        Semaphore-like object for signaling task processing state
```

Comment:
The name and description here are very vague, and it's hard to tell what this is tracking and why without referencing the previous code.  

If I'm not mistaken, all usage of the new _ActiveOperationTracker_ is to replace what previously was done with the service manager's `_is_fulfilling_requirements`  attribute.   Let's rename the param and the attribute and have the docstring say a little more, here and in the other classes/functions where this new type is used.